tests/utils.py

```diff
@@ -351,3 +355,28 @@ def _application_docker_container(
         container.reload()
     yield container
     container.remove(force=True)
+
+
+def find_application_pid(pid: int) -> int:
+    # Application might be run using "sh -c ...", we detect the case and return the "real" application pid
+    process = Process(pid)
+    if (
+        process.cmdline()[0].endswith("sh")
+        and process.cmdline()[1] == "-c"
+        and len(process.children(recursive=False)) == 1
+    ):
+        pid = process.children(recursive=False)[0].pid
+
+    return pid
+
+
+def assert_jvm_flags_equal(actual_jvm_flags: Optional[List], expected_jvm_flags: List) -> None:
```

Comment:
What's the idea? Why is it not a strict `==` comparison? because you don't want to compare the `value` due to ergonomics? Document that next to `pop`ing the items.